01-bubble-sort/src/components/Main/Main.tsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/no-empty-interface */
+
+import React from 'react';
+import { generateArray } from '../../service/generate';
+import { sortStep } from '../../service/sort';
+import { Board } from '../board/Board';
+import { Button } from '../Button/Button';
+import Header from '../Header/Header';
+import { Status } from '../Status/Status';
+import style from './Main.module.scss';
+
+interface IProps {}
+
+interface IState {
+  arrayCell: number[];
+  isStep: boolean;
+  time: number;
+  size: number;
+  isSolved: boolean;
+}
+
+export class Main extends React.Component<IProps, IState> {
+  interval!: NodeJS.Timeout;

```

Comment:
`NodeJS.Timeout` is incorrect. Try `window.setTimeout` explicitely.

Why do you need `!` here? This is not true and incorrect. Just need to define the correct type. `!` should be used very rare and only there where you really need it.